services/alert/service.go

```diff
@@ -717,14 +729,19 @@ func decodeStringToTextUnmarshaler(f, t reflect.Type, data interface{}) (interfa
 func (s *Service) createHandlerFromSpec(spec HandlerSpec) (handler, error) {
 	var h alert.Handler
 	var err error
+	ctx := []keyvalue.T{
+		keyvalue.KV("handler", spec.ID),
+		keyvalue.KV("topic", spec.Topic),
+	}
+	handlerDiag := s.diag.WithHandlerContext(ctx...)
```

Comment:
While redundant should we move this line down the only the cases that use handlerDiag so its not created when its not needed?

Reply:
fixed